docs/birthday-kata.md

```diff
@@ -0,0 +1,890 @@
+
+But maybe we can build a little bit toward the future here. The prompt told us the future will likely include sending birthday messages via other platforms, like SMS or robocalls.
+
+Maybe our model should actually be about a `Message` of which there is some relationship with an `Email`. One possible way to model this would be
+```unison
+unique type Message 
+  = Email EmailAddress Subject Body
+```
+which would give us the option of extending it in the future as needed
+```unison
+unique type Message 
+  = Email EmailAddress Subject Body
+  | SMS PhoneNumber Body
+  | TwitterDM TwitterHandle Body
+```
+
+### Summary
+There isn't one "best" way to model this, but this is one possible way:
+```unison
+unique type Employee = 
+  { id: EmployeeId
+  , name: Name
+  , birthday: Date 
+  }
+
+unique type EmployeeId = EmployeeId Text
+unique type FirstName = FirstName Text
+unique type LastName = LastName Text
+unique type Name = 
+  { first: FirstName
+  , last: LastName
+  }
+
+unique type Date = 
+  { day: Day
+  , month: Month
+  , year: Year
+  }
+unique type Month 
+  = January
+  | February
+  | March
+  | April
+  | May
+  | June
+  | July
+  | August
+  | September
+  | October
+  | November
+  | December
+unique type Day = Day Nat
+unique type Year = Year Nat
+
+unique type Message 
+  = Email EmailAddress Subject Body
+
+unique type EmailAddress = EmailAddress Text
+unique type Subject = Subject Text
+unique type Body = Body Text
+```
+
+## Abilities
+Now that we have our model, let's quickly talk about the effects/infrastructure we know we will need.
+
+One of the goals of this exercise (and generally working as a software engineer) is to decouple infrastructure from our business logic (which is often approximately the same thing as spearating pure and impure code). In OO we would use interfaces to define contracts and dependency inject in whatever implementations we need (be it postgres, aws, in memory, mocks, etc).
+
+In Unison, we leverage Abilities (also called Algebraic Effects) to accomplish something similar. See [the abilities documentation](ability-typechecking.markdown) for a more detailed overview of Abilities.
+
+In our case, let's limit ourselves to the following effects: sending a message, fetching employees from somewhere (a flatfile, a db, an api, anything), and fetching the current date. For the sake of simplicity let's ignore an exception throwing ability (and failures in general) or any other abilities that may make sense to add in a real production system.
+
+```unison
+ability MessageService where
+  send: Message -> Message
+
+ability EmployeeRepository where
+  fetchAllByBirthday: (Day, Month) -> [Employee]
+
+ability Calendar where
+  today: Date
+```
+
+Abilities can sort of be thought of as providing interfaces, and interfaces in turn can be thought of as providing commands.
```

Comment:
I'm not sure what this means 😅. I'm probably missing some context on what "commands" means in this domain, so a note about that might be good for the general readership.

Reply:
Yeah it might just be worth removing. I stole this from the Do Be Do Be Do paper.
```
2.4 Abilities Collect Interfaces; Interfaces Offer Commands
Abilities (Frank’s realisation of algebraic effects) are collections
of parameterised interfaces, each of which describes a choice of
commands (known elsewhere as operations [49]). Command types
may refer to the parameters of their interface but are not otherwise
polymorphic. Here are some simple interfaces
```